cmake/ystdlib-helpers.cmake

```
@@ -0,0 +1,232 @@
+include(CMakePackageConfigHelpers)
+
+# Checks that each argument name is defined and non-empty. Requires a prior call to
+# `cmake_parse_arguments` with `ARG` as the prefix to access the values of the arguments in
+# `REQUIRED_ARG_NAMES`.
+#
+# @param {string[]} REQUIRED_ARG_NAMES
+macro(require_argument_values REQUIRED_ARG_NAMES)
+    set(_REQUIRED_ARGS "${REQUIRED_ARG_NAMES}")
+    foreach(_REQUIRED_ARG IN LISTS _REQUIRED_ARGS)
+        if(NOT DEFINED ARG_${_REQUIRED_ARG} OR ARG_${_REQUIRED_ARG} STREQUAL "")
+            message(FATAL_ERROR "Non-empty value for argument: '${_REQUIRED_ARG}'")
```

I think this error is the opposite of what we mean?
```suggestion
            message(FATAL_ERROR "Empty value for argument: '${_REQUIRED_ARG}'")
```